libr/core/canal.c

```diff
@@ -5404,7 +5404,7 @@ R_API void r_core_anal_esil(RCore *core, const char *str, const char *target) {
 		}
 		r_anal_esil_stack_free (ESIL);
 repeat:;
-	} while (get_next_i (&ictx, &i));
+	} while (get_next_i (&ictx, &i) && i < iend);
```

Comment:
I actually fixed this by doing `cur_addr >= ctx->end_addr` in `get_next_i`. WDYT?

Reply:
Well, in terms of cycles I think your fix wins

Reply:
6dca766a3d431febe664becabd22a1b64c1d8361 ... or you can just close this pr and open your own pr